test/acceptance/DeclarationSchemaSpec.scala

```diff
@@ -0,0 +1,123 @@
+      |  <errors>
+      |     <error>
+      |       <code>xml_validation_error</code>
+      |       <message>cvc-complex-type.3.2.2: Attribute 'foo' is not allowed to appear in element 'Declaration'.</message>
+      |     </error>
+      |  </errors>
+      |</errorResponse>
+    """.stripMargin
+
+  private val apiSubscriptionKeyForXClientIdV1 =
+    ApiSubscriptionKey(clientId = xClientId, context = "customs%2Fdeclarations", version = "1.0")
+  private val apiSubscriptionKeyForXClientIdV2 = apiSubscriptionKeyForXClientIdV1.copy(version = "2.0")
+
+
+  override protected def beforeAll() {
+    startMockServer()
+    stubAuditService()
+    authServiceAuthorizesCSP()
+    startMdgWcoDecService()
+  }
+
+  override protected def afterAll() {
+    stopMockServer()
+  }
+
+  feature("The API handles errors as expected") {
+
+    scenario("Response status 400 when user submits 13 + INV request") {
+      Given("the API is available")
+      val request = ValidSubmission_13_INV_Request.fromCsp.postTo(endpoint)
+
+      When("a POST request with data is sent to the API")
+      val result: Option[Future[Result]] = route(app = app, request)
+
+      Then(s"a response with a 400 status is received")
+      result shouldBe 'defined
+      val resultFuture = result.value
+
+      println(contentAsString(resultFuture))
+      status(resultFuture) shouldBe BAD_REQUEST
+      headers(resultFuture).get(X_CONVERSATION_ID_NAME) shouldBe 'defined
+
+//      And("the response body is a \"invalid xml\" XML")
+//      string2xml(contentAsString(resultFuture)) shouldBe string2xml(BadRequestError)
+    }
+
+
+    scenario("Response status 202 when user submits 13  request") {
+      Given("the API is available")
+      val request = ValidSubmission_13_Request.fromCsp.postTo(endpoint)
+      startApiSubscriptionFieldsService(apiSubscriptionKeyForXClientIdV2)
+      And("the CSP is authorised with its privileged application")
+      authServiceAuthorizesCSP()
+
+      When("a POST request with data is sent to the API")
+      val result: Future[Result] = route(app = app, request).value
+
+      println(contentAsString(result))
+      Then("a response with a 202 (ACCEPTED) status is received")
+      status(result) shouldBe ACCEPTED
+    }
+
+
+    scenario("Response status 400 when user submits INV request") {
+      Given("the API is available")
+      val request = ValidSubmission_INV_Request.fromCsp.postTo(endpoint)
+
+      When("a POST request with data is sent to the API")
+      val result: Option[Future[Result]] = route(app = app, request)
+
+      Then(s"a response with a 400 status is received")
+      result shouldBe 'defined
+      val resultFuture = result.value
+
+      println(contentAsString(resultFuture))
+      status(resultFuture) shouldBe BAD_REQUEST
+      headers(resultFuture).get(X_CONVERSATION_ID_NAME) shouldBe 'defined
+
+//      And("the response body is a \"invalid xml\" XML")
```

Comment:
we might want revisit this or remove comment

Reply:
ok will clean up